src/quantization/q6.jl

```diff
@@ -90,33 +90,74 @@ function quantize!(y::Vector{block_q6_K}, x::Vector{Float32})
     return y
 end
 
-function dequantize!(y::AbstractVector{Float32}, x::AbstractVector{block_q6_K})
-    k = length(y)
-    @assert k % QK_K == 0
-    nb = k ÷ QK_K
-
-    @inbounds for i in 1:nb
-        ql = MutableField(UInt8, x, i, :ql)
-        qh = MutableField(UInt8, x, i, :qh)
-        scales = MutableField(Int8, x, i, :scales)
-        d = Float32(MutableField(Float16, x, i, :d)[1])
-
-        for n in 1:(QK_K÷128)
+# function dequantize!(y::AbstractVector{Float32}, x::AbstractVector{block_q6_K})
+#     k = length(y)
+#     @assert k % QK_K == 0
+#     nb = k ÷ QK_K
+
+#     @inbounds for i in 1:nb
+#         ql = MutableField(UInt8, x, i, :ql)
+#         qh = MutableField(UInt8, x, i, :qh)
+#         scales = MutableField(Int8, x, i, :scales)
+#         d = Float32(MutableField(Float16, x, i, :d)[1])
+
+#         for n in 1:(QK_K÷128)
+#             for l in 1:32
+#                 q1 = reinterpret(Int8, (ql[64*(n-1) + l +  0] & 0xF) | (((qh[32*(n-1) + l] >> 0) & 0x3) << 4)) - 32
+#                 q2 = reinterpret(Int8, (ql[64*(n-1) + l + 32] & 0xF) | (((qh[32*(n-1) + l] >> 2) & 0x3) << 4)) - 32
+#                 q3 = reinterpret(Int8, (ql[64*(n-1) + l +  0]  >> 4) | (((qh[32*(n-1) + l] >> 4) & 0x3) << 4)) - 32
+#                 q4 = reinterpret(Int8, (ql[64*(n-1) + l + 32]  >> 4) | (((qh[32*(n-1) + l] >> 6) & 0x3) << 4)) - 32
+
+#                 is = 8*(n-1) + (l-1) ÷ 16 + 1
+
+#                 y[QK_K*(i-1) + 128*(n-1) + l +  0] = d * (Int16(scales[is + 0]) * q1)
+#                 y[QK_K*(i-1) + 128*(n-1) + l + 32] = d * (Int16(scales[is + 2]) * q2)
+#                 y[QK_K*(i-1) + 128*(n-1) + l + 64] = d * (Int16(scales[is + 4]) * q3)
+#                 y[QK_K*(i-1) + 128*(n-1) + l + 96] = d * (Int16(scales[is + 6]) * q4)
+#             end
+#         end
+#     end
+
+#     return y
+# end
+
```

Comment:
Can probably remove this stuff if it's not needed for now.
```suggestion
```

Reply:
It might still be used for some gguf quantizations, so I would still keep it. The dequantization of token weights is what might use this. https://github.com/cafaxo/Llama2.jl/blob/3d07db9354a3108d2c5dcd8ff5dea2ea9347daf8/src/inference.jl#L198

Reply:
Ah okay, perfect. We should plan on spinning it out at some point as well if it becomes relevant. It may also be worth adding a brief note at the top indicating what it's for and whether it works.